models/issue.go

```diff
@@ -970,6 +970,10 @@ func newIssue(e *xorm.Session, doer *User, opts NewIssueOptions) (err error) {
 		if _, err = e.Exec("UPDATE `milestone` SET num_issues=num_issues+1 WHERE id=?", opts.Issue.MilestoneID); err != nil {
 			return err
 		}
+
+		if _, err = createMilestoneComment(e, doer, opts.Repo, opts.Issue, -1, opts.Issue.MilestoneID); err != nil {
```

Comment:
Sorry, I've seen this only after I approved. The value for `oldMilestoneID` must be 0, not -1.

Reply:
I made -1 too because old code was

https://github.com/go-gitea/gitea/pull/8221/files#diff-d5eea3b2c0d4c92d62e12b34eb8da2c0L1122

Reply:
I see. But when you assign a milestone for an existing issue, the number for that column is 0. That's an inconsistency.

Reply:
done